### PR TITLE
239 utilize `logger::log_shiny_input_change`

### LINKED ISSUE #239: Add logger::log_shiny_input_changes once it allows to log to a custom namespace
The current, latest version of logger does not allow logging log_shiny_input_changes to a custom namespace.

We have a couple of options:
* create our own function logging changes in shiny input (not that hard, just copy logger's with small additions)
* wait for logger to get updated. I have opened a PR into logger with the requested functionality but I cannot be sure when it will get merged. https://github.com/daroczig/logger/pull/93

The objective of the issue is to add the functionality (either our own or logger's) to log changes in shiny inputs. Make sure to exclude sensitive shiny inputs.

### PR DESCRIPTION
**Pull request by m7pr**
Close #239 

The example loading for `tm_a_pca` on a startup shows how the input change logging is made. 

We are connected to package namespace (`namespace = "teal.modules.general"`) 
that is created on a startup with `teal.logger::register_logger(namespace = "teal.modules.general")`

This function needs to be called in the server function, so there is no way this can be run during the package startup.

image

**Note:** Needed to bring `logger` to Imports.

### CODE REVIEW

**Comment by m7pr** [comment]
@pawelru kindly asking you for the review, as you are the logging expert these days :) 
I needed to move logger to Imports. Do you think there is a way to extend `teal.logger::register_logger` to utilize this so `logger` can stay in Suggests? `logger::log_shiny_input_changes` needs to be called in `server` function.

**Comment by pawelru** [comment]
> Do you think there is a way to extend `teal.logger::register_logger` to utilize this so `logger` can stay in Suggests? `logger::log_shiny_input_changes` needs to be called in `server` function.

There was a discussion about this here: https://github.com/insightsengineering/teal.logger/issues/74. This idea collapses for some good reasons. You should not be afraid of promoting a package if it is really used. I'm personally not a big fan of hiding (true) dependencies. I think it creates more bad than good. What you have done is correct.

**Comment by m7pr** [comment]
Got it, thanks @pawelru for suggestions

File: DESCRIPTION
[the remark this answers is not in the document]
Comment (m7pr) [reply DESCRIPTION:1]: 2399cbde680e8e9a4f17bdf85d2dec31db16e0c2

[the remark this answers is not in the document]
Comment (m7pr) [reply DESCRIPTION:1]: Yeah, great advice. Thanks db0e840ec9590ab8e7e7888dcc7729198db32816

**Comment by m7pr** [comment]
Thanks @pawelru - I did updated NEWS, verified content of `pre-commit-config.yaml` and part of DESCRIPTION for `Config/verdepcheck` in this PR and others as well. Would you accept? 

https://github.com/insightsengineering/teal.goshawk/pull/274
https://github.com/insightsengineering/teal.osprey/pull/265
https://github.com/insightsengineering/teal.modules.hermes/pull/382
https://github.com/insightsengineering/teal.modules.clinical/pull/1182

**Comment by pawelru** [comment]
Thanks. It looks to me that changes in `logger` you proposed can go its own way. After adding the `if` statement this is independent. I think it's good to go and after that PR we can remove the if statement (and vbump dep)

**Review by pawelru — APPROVED** [review]

**Comment by m7pr** [comment]
Created an issue, for the future, to keep in mind that the `if` statement might not be needed anymore at some point https://github.com/insightsengineering/teal.modules.general/issues/758

**Pull request merged**